src/test/ReactTestUtils.js

```diff
@@ -384,6 +384,44 @@ ReactShallowRenderer.prototype.unmount = function() {
   }
 };
 
+ReactShallowRenderer.prototype.findElementByRef = function(refName) {
+  var result = null;
+  var renderedElement = this.getRenderOutput();
+
+  if (null !== renderedElement) {
+    result = this._findRef(refName, renderedElement);
+  }
+
+  return result;
+};
+
+ReactShallowRenderer.prototype._findRef = function(refName, element) {
+  if (element.ref == refName) {
+    return element;
+  }
+
+  var children = element.props.children;
+
+  if (typeof children == "object") {
+
+    if (typeof children.forEach == "function") {
+      var result = null;
+
+      children.forEach(function(element) {
+        result = this._findRef(refName, element) || result;
```

Comment:
I know i should quit foreach when element's found but Array.prototype.forEach does not support breaks. Array.prototype.some is able to do this but I'm not sure it is safe